src/renderer/qssWidget/js/qssOfficeWidget.js

```diff
@@ -0,0 +1,391 @@
+                    stateNames.push(name);
+                }
+                allStates++;
+            }
+        });
+
+        if (stateNames.length === 0) {
+            // no option is selected
+            return allFalse;
+        } else if (stateNames.length === allStates) {
+            // all of the options are selected
+            return allTrue;
+        } else if (stateNames.length === 1) {
+            // only one of the options is selected
+            // IMPORTANT: returning the name of the option as command
+            return stateNames.pop();
+        } else {
+            // IMPORTANT: we should never got to here
+            // but just in case, returning the default command
+            return defaultCommand;
+        }
+    };
+
+    /**
+     * Pre-loads the data in the office.model.states array
+     * @param {Component} that - The `gpii.qssWidget.office.presenter` instance.
+     * @param {Component} office - The `gpii.qssWidget.office` instance.
+     * @param {String} ribbonState - The initial state of ribbons
+     */
+    gpii.qssWidget.office.loadState = function (that, office) {
+        // pre-fills the states of all available schema keys
+        if (office.model.setting.value === office.model.availableCommands.allTrueCommand) {
+            fluid.each(office.model.setting.schema.keys, function (key) {
+                if (key !== office.model.availableCommands.resetCommand) {
+                    office.model.states[key] = true;
+                }
+            });
+        } else if (office.model.setting.value === office.model.availableCommands.allFalseCommand) {
+            fluid.each(office.model.setting.schema.keys, function (key) {
+                if (key !== office.model.availableCommands.resetCommand) {
+                    office.model.states[key] = false;
+                }
+            });
+        } else {
+            fluid.each(office.model.setting.schema.keys, function (key) {
+                if (key === office.model.setting.value && key !== office.model.availableCommands.resetCommand) {
+                    office.model.states[key] = true;
+                }
+            });
+        }
+
+        // // checks the checkboxes if needed
+        gpii.qssWidget.office.applyCheckmarks(that, office.model.states) ;
+
+    };
+
+    /**
+     * Adds a checked icon next to a setting option if it is the currently selected one for the setting.
+     * @param {Component} that - The `gpii.qssWidget.presenter` instance.
+     * @param {Component} states - simple true/false object with the current states
+     */
+    gpii.qssWidget.office.applyCheckmarks = function (that, states) {
+        if (states[that.model.item.key] === true) {
+            that.container.attr("aria-checked", true);
+        }
+    };
+
+    gpii.qssWidget.office.updateValue = function (that, key, item, container, office, repeater) {
+
+        // toggle the current state
+        office.model.states[key] = !office.model.states[key];
```

Comment:
Always assign into the model by using the ChangeApplier

Reply:
Done, same as in loadState.